### PR TITLE
fix(tiflash): use hub.pingcap.net/tiflash/tiflash-llvm-base:rocky8-llvm-17.0.6 as build image

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
The same as CD.

### CODE REVIEW

File: pipelines/pingcap/tiflash/latest/pod-merged_build.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ spec:
     fsGroup: 1000
   containers:
     - name: runner
-      image: "hub.pingcap.net/tiflash/tiflash-llvm-base:llvm-17.0.6-rocky8"
+      image: "hub.pingcap.net/tiflash/tiflash-llvm-base:rocky8-llvm-17.0.6"
```
Comment (wuhuizuo) [inline pipelines/pingcap/tiflash/latest/pod-merged_build.yaml:8]: the toolchains in the images maybe different, maybe the CI needs more tools in the flow than CD flow.

**Comment by wuhuizuo** [comment]
@purelind please check the toolchains in the new image.

**Comment by purelind** [comment]
> @purelind please check the toolchains in the new image.

What is the difference between these two images?

**Comment by Lloyd-Pottiger** [comment]
> > @purelind please check the toolchains in the new image.
> 
> What is the difference between these two images?

hub.pingcap.net/tiflash/tiflash-llvm-base:rocky8-llvm-17.0.6 is maintained in TiFlash repo. The two images are nearly the same, there is no need to maintain hub.pingcap.net/tiflash/tiflash-llvm-base:llvm-17.0.6-rocky8 again.

**Comment by purelind** [comment]
> > > @purelind please check the toolchains in the new image.
> > 
> > 
> > What is the difference between these two images?
> 
> hub.pingcap.net/tiflash/tiflash-llvm-base:rocky8-llvm-17.0.6 is maintained in TiFlash repo. The two images are nearly the same, there is no need to maintain hub.pingcap.net/tiflash/tiflash-llvm-base:llvm-17.0.6-rocky8 again.

Okay, I will compare the differences between the two, conduct a retest current tiflash pipelines, and merge after passing the test.

**Comment by purelind** [comment]
retest link
* integration test https://do.pingcap.net/jenkins/job/pingcap/job/tiflash/job/pull_integration_test/2131/
* unit test https://do.pingcap.net/jenkins/job/pingcap/job/tiflash/job/pull_unit_test/2180/

**Comment by wuhuizuo** [comment]
> retest link
> 
> * integration test https://do.pingcap.net/jenkins/job/pingcap/job/tiflash/job/pull_integration_test/2131/
> * unit test https://do.pingcap.net/jenkins/job/pingcap/job/tiflash/job/pull_unit_test/2180/

All are failed.

We will only publish TiFlash binaries from CD building to fileserver rather than CI pipelines.

File: pipelines/pingcap/tiflash/latest/pod-merged_build.yaml (hunk 1 above)
Comment (purelind) [reply pipelines/pingcap/tiflash/latest/pod-merged_build.yaml:8]: Missing tool: hostname

**Pull request closed without merging**